File: fixtures/go/ledger.go
package ledger

import (
	"errors"
	"fmt"
	"strings"
)

type Entry struct {
	Title   string
	Tags    []string
	Settled bool
}

func Parse(raw string) (Entry, error) {
	parts := strings.Split(raw, "#")
	title := strings.TrimSpace(parts[0])
	if title == "" {
		return Entry{}, errors.New("title is required")
	}
	return Entry{Title: title, Tags: parts[1:]}, nil
}

func (entry Entry) Summary() string {
	return fmt.Sprintf("%s [%s]", entry.Title, strings.Join(entry.Tags, ", "))
}

File: fixtures/rust/ledger.rs
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct Entry {
    pub title: String,
    pub tags: Vec<String>,
    pub settled: bool,
}

impl Entry {
    pub fn parse(raw: &str) -> Result<Self, &'static str> {
        let mut parts = raw.split('#').map(str::trim);
        let title = parts.next().unwrap_or_default();
        if title.is_empty() {
            return Err("title is required");
        }
        Ok(Self {
            title: title.to_owned(),
            tags: parts.filter(|tag| !tag.is_empty()).map(str::to_owned).collect(),
            settled: false,
        })
    }
}
